Resharers now visible on top of owner entry

PVince81 opened the pull request:
## Description
For received shares, the resharers user display names are now shown
on top of the owner entry in the collaborators list, with a reshare
icon, instead of having their own entry in the collaborators list.

Moved reshare related tests to the reshare feature file.

## Related Issue
- Fixes https://github.com/owncloud/enterprise/issues/3850

## Motivation and Context
Based on feedback after reviewing the feature.

## How Has This Been Tested?
- manual test
- acceptance test

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in https://github.com/owncloud/documentation -->
- [x] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

kulmann commented on apps/files/src/components/FileSharingSidebar.vue:
```diff
@@ -153,58 +151,40 @@ export default {
+        info: {
+          path: ownerShare.info.path,
+          // set to user share for displaying user
+          share_type: '' + shareTypes.user, // most code requires string..
```
We recently spent some time into providing integers where possible, but not in `collaborator.info.share_type`, but in `collaborator.shareType`.

PVince81 replied:
I killed all usage of "info" now as I grew tired of it, so this is gone as well

kulmann commented:
![Bildschirmfoto 2020-02-21 um 13 09 04](https://user-images.githubusercontent.com/3532843/75034077-abb38b00-54ac-11ea-847c-b216bd4619e6.png)
![Bildschirmfoto 2020-02-21 um 13 08 45](https://user-images.githubusercontent.com/3532843/75034076-ab1af480-54ac-11ea-9118-57d3fcc4c265.png)

I like the `xsmall` icon for the resharer better than the `small` one.

LukasHirt commented:
Honestly feels a bit odd to me seeing there twice the same user in case of the same owner and sharer.

![image](https://user-images.githubusercontent.com/25989331/75035985-136bd500-54b1-11ea-8d39-c964958e1874.png)

PVince81 commented:
> Honestly feels a bit odd to me seeing there twice the same user in case of the same owner and sharer.

bug in this PR

PVince81 commented:
- Added smaller icon
- Fixed bug about resharer appearing as the same name
- Refactored share object
- Added popup when clicking on the reshare info:
<img width="297" alt="image" src="https://user-images.githubusercontent.com/277525/75045055-e6282280-54c2-11ea-82d4-8e71c2d5bdb7.png">

- [ ] tweak the popup styles

PVince81 commented on apps/files/src/components/FileSharingSidebar.vue:
```diff
@@ -153,58 +151,40 @@ export default {
-      ]
-    },
-    $_shareOwnerAsCollaborator () {
+    $_ownerShare () {
```
done

PVince81 commented:
@kulmann @LukasHirt please re-review

I grew tired of `.info` so now have reduced its usage.

PVince81 commented:
- [ ] change popup text to "Shared with you by"
- [ ] move popup position, maybe down a bit so the owner row stays fully visible

kulmann commented on apps/files/src/components/Collaborators/Collaborator.vue:
```diff
@@ -4,8 +4,30 @@
+            <ul class="uk-list uk-list-divider uk-overflow-hidden uk-margin-remove">
+              <li v-for="resharer in collaborator.resharers" :key="resharer.name">
+                <div class="uk-flex uk-flex-middle uk-flex-left">
+                  <avatar-image v-if="collaborator.shareType === shareTypes.user" class="uk-margin-small-right" :width="48" :userid="resharer.name" :userName="resharer.displayName" />
```
`v-if` part doesn't make sense. We want to show an avatar if the shareType of the **resharer** is user. shareType of the **owner** is irrelevant.

kulmann replied:
resharer object doesn't contain the shareType information so far.

PVince81 replied:
right, this was a copy-paste

LukasHirt commented on apps/files/src/components/Collaborators/Collaborator.vue:
```diff
@@ -4,8 +4,30 @@
+            ref="menu"
+            closeOnClick
+          >
+            <h4><translate>Reshared by:</translate></h4>
```
Can be made shorter with:

`<h4 v-translate>Reshared by:</h4>`

or

`<translate tag="h4">Reshared by:</translate>`

PVince81 commented:
PR adjusted according to comments and checkboxes

LukasHirt commented on apps/files/src/components/Collaborators/Collaborator.vue:
```diff
@@ -3,9 +3,33 @@
-          <oc-icon name="repeat" class="uk-preserve-width" />
-          <span class="uk-padding-remove uk-margin-xsmall-left uk-text-truncate files-collaborators-collaborator-reshare-information">{{ $_reshareInformation }}</span>
+        <div class="uk-text-meta">
+          <a :id="$_resharerToggleId" :aria-label="$gettext('Show resharer details')" class="uk-flex uk-flex-middle">
```
We have the `raw` variation of the button where the styling is the same as a link, right? Use that instead pls

PVince81 replied:
the raw variation still capitalizes the text :-/

LukasHirt replied:
I see. I'd see that as an issue to be raised in ODS.

LukasHirt replied:
Ah, we already have one - https://github.com/owncloud/owncloud-design-system/issues/442

PVince81 replied:
adjusted to use oc-button

however we'll need to adjust ODS to also not force a height on raw buttons: https://github.com/owncloud/owncloud-design-system/issues/654

LukasHirt commented on apps/files/src/components/Collaborators/Collaborator.vue:
```diff
@@ -3,9 +3,33 @@
-          <oc-icon name="repeat" class="uk-preserve-width" />
-          <span class="uk-padding-remove uk-margin-xsmall-left uk-text-truncate files-collaborators-collaborator-reshare-information">{{ $_reshareInformation }}</span>
+        <div class="uk-text-meta">
+          <a :id="$_resharerToggleId" :aria-label="$gettext('Show resharer details')" class="uk-flex uk-flex-middle">
```
Shouldn't it be displayed only if there are any resharers?

PVince81 replied:
the check is done higher above on the table row

LukasHirt replied:
I see now - overlooked it before. Nice, thanks!

PVince81 commented:
- [ ] REQUIRES: https://github.com/owncloud/owncloud-design-system/issues/654 to not look shitty

PVince81 commented:
- [ ] use OcDisclosureDrop instead of OcDrop as the popup is not a menu

PVince81 commented:
Possibly related failures:
- tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature:63
- tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature:332
- tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature:509
- tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature:520
- and many resharer scenarios, tests/acceptance/features/webUIResharing/reshareUsers.feature:13 being one of them

PVince81 commented:
fixed the test issue: the selector for the collaborator name was confused because the reshare popup was using the same class name. Fixed by using a different class name in the popup.

PVince81 commented:
Remaining failure:
- tests/acceptance/features/webUIResharing/reshareUsers.feature:144

PVince81 commented:
Adjusted. Expecting CI to pass.

Then it's done expect for the extra height due to missing ODS update.

PVince81 commented:
rebased to get the ODS update and restart the build due to setup failure

LukasHirt commented on apps/files/src/components/Collaborators/Collaborator.vue:
```diff
@@ -3,9 +3,35 @@
-          <oc-icon name="repeat" class="uk-preserve-width" />
-          <span class="uk-padding-remove uk-margin-xsmall-left uk-text-truncate files-collaborators-collaborator-reshare-information">{{ $_reshareInformation }}</span>
+        <div class="uk-text-meta">
+          <oc-button type="a" variation="raw" :id="$_resharerToggleId" :aria-label="$gettext('Show resharer details')">
```
Without the `type` "a" pls. It's important to keep the button in here for semantics and accessibility.

PVince81 replied:
why keep the type attribute at all then ? should we kill it if we cannot use it ?

PVince81 replied:
adjusted and retested, it looks the same

PVince81 commented:
@LukasHirt all comments addressed, can we merge ?
